### PR TITLE
Docs: Integrations - Adds an intro page

### PR DESCRIPTION
**Pull request by kiwicopple**

### CODE REVIEW

File: apps/reference/docs/guides/integrations/integrations.mdx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+### Listing your integration
+
+You can list your integration in our Partners marketplace and inside the Supabase Docs by [applying](https://supabase.com/partners/integrations#become-a-partner) through our Partners program.
+
+Integrations will be assessed on the following criteria:
+
+- **Business viability**:<br />while we welcome everyone to built an integration, we will only list companies that are deemed to be long-term viable. This includes an official business registration and bank account, meaningful revenue, or VC backing. We require this criteria to ensure the health of the marketplace.
+- **Compliance**:<br />Integrations should not infringe on Supabase brand/trademark. In short, you cannot use "Supabase" in the name. Because the listing will appear on the Supabase domain, we don't want to mislead developers into thinking that an integration is an official product.
+- **Sevice Level Agreements**:<br />All listings are required to have their own Terms and Conditions, Privacy Policy, and Acceptable Use Policy, and the company must have resources to meet their SLAs.
```
Comment (thorwebdev) [inline apps/reference/docs/guides/integrations/integrations.mdx:29]:
```suggestion
- **Service Level Agreements**:<br />All listings are required to have their own Terms and Conditions, Privacy Policy, and Acceptable Use Policy, and the company must have resources to meet their SLAs.
```

File: apps/reference/nav/_referenceSidebars.js [hunk 2]
```diff
@@ -215,23 +215,50 @@ const sidebars = {
...
+            'guides/integrations/auth0',
+            'guides/integrations/authsignal',
+            'guides/integrations/clerk',
+            'guides/integrations/stytch',
+            'guides/integrations/supertokens',
+          ],
+        },
+        {
+          type: 'category',
+          label: 'Offline/Caching',
```
Comment (thorwebdev) [inline apps/reference/nav/_referenceSidebars.js:233]:
```suggestion
          label: 'Caching / Offline-First',
```

File: apps/reference/nav/_referenceSidebars.js [hunk 3]
```diff
@@ -215,23 +215,50 @@ const sidebars = {
...
+            'guides/integrations/pgmustard',
+            'guides/integrations/prisma',
+            'guides/integrations/sequin',
+            'guides/integrations/snaplet',
+            'guides/integrations/vercel',
+          ],
+        },
+        {
+          type: 'category',
+          label: 'Low Code',
```
Comment (thorwebdev) [inline apps/reference/nav/_referenceSidebars.js:251]:
```suggestion
          label: 'Low-Code',
```

**Review by thorwebdev — APPROVED** [review]
LGTM, thanks. @dannykng you good with this also?

**Review by dannykng — APPROVED** [review]

**Pull request merged**